doc/api.md

```diff
@@ -742,6 +744,60 @@ Unreserves a path for a publishing application. Returns success or failure only.
 - If a path reservation exists for the supplied base_path but a different
   publishing_app, the command will fail.
 
+## `PUT /publish-intent(/*base_path)`
+
+[Request/Response detail][put-intent-pact]
+
+Used to create or update a publishing intent for a particular base_path and
+reserve the base path in the publishing-api. A publishing intent sets when the TTL
+cache should be cleared.
+
+### Path parameters
+
+- `base_path`
+  - Identifies the path to create the publishing intent for
+
+### JSON parameters:
+
+- `publishing_app` *(required)*
+  - The name of the application making this request, words separated with
+    hyphens.
+- `rendering_app` *(required)*
+  - The name of the application rendering the content the base path refers to,
+    words separated with hyphens.
+- `publish_time` *(required)*
+  - The `DateTime` the content is set to be published.
+- `routes` *(optional)*
+  - `Array` - Fields supported are path and type. `path` can either be an exact
```

Comment:
This also looks helpful, but I notice the other docs don't go into this much detail. It might be more future-proof to just link to the [definition in the content schemas](https://github.com/alphagov/govuk-content-schemas/blob/master/formats/shared/definitions/routes_redirects.jsonnet), which I'm guessing is what it's validating against.

Reply:
Perhaps it would be better to indicate why this is optional.

Reply:
Yep I can do both of those things, was unsure about how much detail to add here!